### DISCUSSION TITLE
http response status for void methods

### DISCUSSION THREAD

**Discussion by andruskutt** [micronaut-projects/micronaut-core#5665] [Ideas]
I have in controller method with return type void. Current returned http status for it is 200 OK.
Maybe it's better to return 204 NO CONTENT for these methods by default?
I know I can use @Status(HttpStatus.NO_CONTENT) but this seems double work when we have declared method without return value.

**Comment by jameskleeh** [comment]
I agree from the narrow perspective that this makes sense but there are also cases where data is returned even though the method returns void. For example with the `@View` annotation. Also there is the case of error routes that may return void that wouldn't make sense to return a 204 for. I think keeping the rule simple that the default status is 200 unless changed makes things easier to reason about
